src/autorest-core/lib/message.ts

```diff
@@ -30,7 +30,10 @@ export interface Message {
   Key?: Iterable<string>;
   Details?: any;
   Text: string;
+
+  // injected or modified by core
   Source?: Array<SourceLocation>;
   Range?: Iterable<Range>;
   Plugin?: string;
+  FormattedMessage?: string;
```

Comment:
Thanks for making this change. Side question, would it make sense to have another property like "InfoUrl" in the Message, so it can carry a link to a wiki or similar that contains more information on the issue, or is that expected to be somehow included the Details?